src/features/organizations/hooks/useConnectOrg.ts

```diff
@@ -0,0 +1,34 @@
+import { useApiClient, useAppDispatch } from 'core/hooks';
+import { userMembershipsLoaded } from '../store';
+import { ZetkinMembership, ZetkinPersonNativeFields } from 'utils/types/zetkin';
+import useFollowOrgMutations from './useFollowOrgMutations';
+
+export default function useConnectOrg(orgId: number) {
+  const apiClient = useApiClient();
+  const dispatch = useAppDispatch();
+  const { followOrg } = useFollowOrgMutations(orgId);
+
+  const connectOrg = async () => {
+    const person = await apiClient.post<ZetkinPersonNativeFields>(
+      `/api/orgs/${orgId}/join_requests`,
+      {}
+    );
+    if (person.id) {
+      const memberships = await apiClient.get<ZetkinMembership[]>(
+        `/api/users/me/memberships`
+      );
+      dispatch(userMembershipsLoaded(memberships));
+      if (memberships) {
+        const newMembership =
+          memberships.find(
+            (membership) => membership.organization.id == orgId
+          ) || null;
+        if (newMembership) {
+          followOrg(newMembership);
+        }
+      }
+    }
```

Comment:
Maybe this should be moved to an RPC, since this single operation requires three API requests?